nimble/host/src/ble_gap.c

```diff
@@ -1806,15 +1806,40 @@ int
 ble_gap_rx_l2cap_update_req(uint16_t conn_handle,
                             struct ble_gap_upd_params *params)
 {
+    struct ble_gap_upd_params self_params;
     struct ble_gap_event event;
     int rc;
 
+    /* Copy the peer params into the self params to make it easy on the
+     * application.  The application callback will change only the fields which
+     * it finds unsuitable.
+     */
+    self_params = *params;
+
     memset(&event, 0, sizeof event);
     event.type = BLE_GAP_EVENT_L2CAP_UPDATE_REQ;
     event.conn_update_req.conn_handle = conn_handle;
+    event.conn_update_req.self_params = &self_params;
     event.conn_update_req.peer_params = params;
 
     rc = ble_gap_call_conn_event_cb(&event, conn_handle);
+
+    if ((self_params.itvl_min <= self_params.itvl_max)
+         && (self_params.itvl_min >= params->itvl_min)
+         && (self_params.itvl_max <= params->itvl_max)) {
+             params->itvl_max = self_params.itvl_max;
+             params->itvl_min = self_params.itvl_min;
+    }
+
+    if ((self_params.min_ce_len <= self_params.max_ce_len)
```

Comment:
Unless I'm mistaken we probably shouldn't restrict the min/max_ce_len values as they are not requested by the peer and actually just initialized [here](https://github.com/apache/mynewt-nimble/blob/6ced1dd4ca63e6f0bd2df7dcc558afcc2896f1c2/nimble/host/src/ble_l2cap_sig.c#L414) to:
```
    params.min_ce_len = BLE_GAP_INITIAL_CONN_MIN_CE_LEN;
    params.max_ce_len = BLE_GAP_INITIAL_CONN_MAX_CE_LEN;
```
before this gets called.

Reply:
I checked in a fix, but realized I might have done the opposite thing to what you wanted.  Do you want to allow the callback to change the ce_len to whatever it wants, or do you want to prevent it from changing them at all?  I did the latter, but I think you may have meant the former.

Reply:
Yes, I think the app should be able to change the `min/max_ce_len` to whatever it wants. 
That's just my opinion, we'll have to see what @rymanluk thinks.

Also, since you're using the esp32 as well you might be interested in [this](https://github.com/espressif/esp-nimble/issues/4) for some insight into my opinion.